scala/scala.bzl

```diff
@@ -362,8 +366,8 @@ def _collect_jars(targets):
             # see JavaSkylarkApiProvider.java,
             # this is just the compile-time deps
             # this should be improved in bazel 0.1.5 to get outputs.ijar
```

Comment:
Reverted that part of the change. Amended comment.

My fear is that transitive_deps to ijar COULD break code (although technically code that was making incorrect assumptions anyway). However, I was hoping for this PR to strictly decrease number of breakages.